### PR TITLE
feat(AvatarGenerated): add background prop and rounded aspect

### LINKED ISSUE #130: AvatarGenerated should have rounded aspect and has a transparent background
Screen Shot 2022-08-30 at 12 08 51 AM

### LINKED ISSUE #132: Add `brand` color to theme
### Motivation

Create a centralized place to have `fuel` color: `#58c09b`

### PR DESCRIPTION
**Pull request by LuizAsFight**
Changes to `AvatarGenerated` component:
- add rounded aspect
- add `background` prop
  - undefined (default) -> transparent will be applied
  - `fuel` -> will use Fuel green color
  - `random` -> will generate a color from the `hash` input
  - string -> fixed color to use in the background

Add `brand` color to theme = `#58c09b`

Closes #130 
Closes #132

### CODE REVIEW

File: design-system/react/src/components/Avatar/utils.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const getBackgroundColor = (
+  backgroundColor?: string,
+  unique?: string
+): string | undefined => {
+  if (backgroundColor === 'fuel') return '#58c09b';
+  if (backgroundColor !== 'random') return backgroundColor;
+  if (!unique) return unique;
+
```
Comment (luizstacio) [inline design-system/react/src/components/Avatar/utils.ts:8]:
```suggestion
  backgroundColor?: 'fuel' | 'random' | string
): string | undefined => {
  if (backgroundColor === 'fuel') return '#58c09b';
  if (backgroundColor !== 'random') return backgroundColor;

```

File: design-system/react/src/components/Avatar/utils.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const getBackgroundColor = (
+  backgroundColor?: string,
+  unique?: string
+): string | undefined => {
+  if (backgroundColor === 'fuel') return '#58c09b';
```
Comment (luizstacio) [inline design-system/react/src/components/Avatar/utils.ts:5]: Can we get the color from the theme?

File: design-system/react/src/components/Avatar/utils.ts (hunk 1 above)
Comment (luizstacio) [reply design-system/react/src/components/Avatar/utils.ts:8]: Can we remove the unique prop?

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: design-system/react/src/components/Avatar/utils.ts (hunk 1 above)
Comment (LuizAsFight) [reply design-system/react/src/components/Avatar/utils.ts:8]: `unique` is being directly used to generate the color. It is basically the wallet address or any unique hash informed via `hash` prop in `AvatarGenerated` component.

Comment (luizstacio) [reply design-system/react/src/components/Avatar/utils.ts]: maybe just change for hash to follow the same name?

**Review by pedronauck — CHANGES_REQUESTED** [review]

File: design-system/react/src/components/Avatar/AvatarGenerated.tsx [hunk 3]
```diff
@@ -16,15 +18,20 @@ export type AvatarGeneratedProps = BoxProps & {
...
 
 export const AvatarGenerated = createComponent<
   AvatarGeneratedProps,
   unknown,
   OmitProps
->(({ hash, size = "md", className, css, as, ...props }) => {
+>(({ hash, size = "md", className, css, as, background, ...props }) => {
   const totalSize = SIZES[size];
-  const svgString = toSvg(hash, totalSize);
+  const backColor = getBackgroundColor(background, hash);
```
[the remark this answers is not in the document]
Comment (pedronauck) [reply design-system/react/src/components/Avatar/AvatarGenerated.tsx:30]: Some tests here will be good to keep our coverage 😅

**Review by pedronauck — APPROVED** [review]

File: design-system/react/src/components/Avatar/AvatarGenerated.test.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    const { container } = await render(
+      <Avatar.Generated hash="0x2cab234f35a4" background="random" />
+    );
+
+    const svg = container.querySelector("svg");
+    expect(svg).toBeInTheDocument();
+
+    const rect = container.querySelector("svg rect");
+    expect(rect).toBeInTheDocument();
+    expect(rect?.getAttribute("fill")).not.toEqual("");
```
Comment (luizstacio) [inline design-system/react/src/components/Avatar/AvatarGenerated.test.tsx:36]: It's not possible to get the svg output string? To check if the svg is always equal.

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**